lib/src/visualisation.dart

```diff
@@ -85,9 +88,13 @@ class Visualisation {
     if (width != -1) {
       image = copyResize(image, width, height, NEAREST);
     }
-    new io.File(_imagePrefix + new DateTime.now().toString() + '.png')
-        .create(recursive: true)
-        .then((io.File file) {
+    String filename = _outputFolder;
+    if (_outputFilename == null) {
+      filename = filename + new DateTime.now().toString() + '.png';
```

Comment:
Writing this as filename = "$_outputFolder${DateTime.now().png" would be a little more darty, and has some nice null safety properties.